client/lib/memoize-last/index.ts

```diff
@@ -24,3 +24,21 @@ export default function memoizeLast< T extends ( ...args: any[] ) => any >( fn:
 		return lastResult;
 	} ) as T;
 }
+
+/**
+ * A stricter-typed alias of `memoizeLast`, for functions without arguments.
+ * Since it only accepts functions without arguments, it effectively guarantees
+ * that the provided function will only be run once, as the check for whether
+ * the arguments have not changed will always pass.
+ *
+ * @param fn The function to be wrapped.
+ *
+ * @returns The wrapped function.
+ */
+export function once< T extends () => any >( fn: T ) {
+	// Runtime validation. Useful when static validation is unavailable.
+	if ( fn.length !== 0 ) {
+		throw new Error( 'memoize-last: The `once` method expects a function with no arguments.' );
```

Comment:
Such assertions are often limited only to `process.env.NODE_ENV !== 'production'`.

Reply:
Yes, I think that makes a lot of sense. Thanks! Change applied.